templates/common/aws/files/usr-local-bin-aws-kubelet-nodename.yaml

```diff
@@ -14,10 +14,9 @@ contents:
 
     # For compatibility with the AWS in-tree provider
     # Set node name to be instance name instead of the default FQDN hostname
-    #
-    # https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instancedata-data-retrieval.html
-    name=$(curl -s http://169.254.169.254/2016-09-02/meta-data/local-hostname)
+    # afterburn service is using for metadata retrival, see respective systemd unit
+    # metadata related afterburn doc: https://coreos.github.io/afterburn/usage/attributes/
     cat > "${NODECONF}" <<EOF
     [Service]
-    Environment="KUBELET_NODE_NAME=${name}"
+    Environment="KUBELET_NODE_NAME=${AFTERBURN_AWS_HOSTNAME}"
```

Comment:
This is the content of `/meta-data/hostname` though, while before it was `local-hostname`. Not sure if there are semantic differences between the two.

Reply:
https://docs.aws.amazon.com/AWSEC2/latest/UserGuide/instancedata-data-categories.html - doc description of `local-hostname` and `hostname` says the same, as far as i can judge

checked it on one of the nodes
```
[root@ip-10-0-138-119 ~]# curl -s http://169.254.169.254/2016-09-02/meta-data/hostname -w "\n"
ip-10-0-138-119.ec2.internal

[root@ip-10-0-138-119 ~]# curl -s http://169.254.169.254/2016-09-02/meta-data/local-hostname -w "\n"
ip-10-0-138-119.ec2.internal
```